File: c/tree/minimum-depth-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int minDepth(struct TreeNode* root) {
    if(root == NULL)
        return 0;

    if(root->left == NULL && root->right == NULL)
        return 1;

    int left = minDepth(root->left);
    int right = minDepth(root->right);

    if(left == 0)
        left = INT_MAX;
    if(right == 0)
        right = INT_MAX;

    return left < right ? left + 1 : right + 1;
}

File: python/tree/minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def minDepth(self, root):
        if root is None:
            return 0

        left = self.minDepth(root.left)
        right = self.minDepth(root.right)

        if left == 0 and right == 0:
            return 1
        elif left == 0:
            return right + 1
        elif right == 0:
            return left + 1
        else:
            return min(left, right) + 1
